spec/controllers/users_controller_spec.rb

```diff
@@ -1,347 +1,318 @@
+# frozen_string_literal: true
 require 'spec_helper'
 
-shared_examples_for 'page success' do
-  it { is_expected.to respond_with(:success) }
-  it { is_expected.not_to set_flash }
-end
-
-shared_examples_for 'access denied' do
-  it { is_expected.to redirect_to(root_url) }
-  it { is_expected.to set_flash }
-end
-
 describe UsersController, type: :controller do
-  before(:each) { @ac = mock_app_config }
-  let!(:user) { FactoryGirl.create(:user) }
+  before(:each) { mock_app_config }
 
-  context 'with admin user' do
-    before do
-      sign_in FactoryGirl.create(:admin)
-    end
-    describe 'GET index' do
-      let!(:banned) { FactoryGirl.create(:banned) }
-      let!(:other_user) { FactoryGirl.create(:user) }
+  it_behaves_like 'calendarable', User
+
+  describe 'GET index' do
+    before { mock_user_sign_in(UserMock.new(:admin)) }
+    context 'basic function' do
       before { get :index }
-      it_behaves_like 'page success'
-      it { is_expected.to render_template(:index) }
-      context 'without show banned' do
-        it 'should assign users to all active users' do
-          expect(assigns(:users).include?(other_user)).to be_truthy
-          expect(assigns(:users).include?(banned)).to be_falsey
-        end
-      end
-      context 'with show banned' do
-        before { get :index, show_banned: true }
-        it 'should assign users to all users' do
-          expect(assigns(:users).include?(banned)).to be_truthy
-        end
-      end
+      it_behaves_like 'successful request', :index
+    end
+    it 'defaults to active users' do
+      allow(User).to receive(:active).and_return(User.none)
+      get :index
+      expect(User).to have_received(:active)
     end
-    describe 'GET show' do
-      before { get :show, id: user }
-      it_behaves_like 'page success'
-      it { is_expected.to render_template(:show) }
 
-      context 'with banned user' do
-        before { get :show, id: FactoryGirl.create(:banned) }
-        it { is_expected.to respond_with(:success) }
-        it { is_expected.to set_flash }
-      end
+    it 'orders users by username' do
+      allow(User).to receive(:active).and_return(User)
+      allow(User).to receive(:order)
+      get :index
+      expect(User).to have_received(:order).with('username ASC')
     end
 
-    describe 'POST quick_new' do
-      context 'possible netid provided' do
-        before { post :quick_new, format: :js, possible_netid: 'csw3' }
-        it 'should assign @user to the possible netid' do
-          expect(assigns(:user).attributes).to\
-            eq(User.new(User.search_ldap('csw3')).attributes)
-        end
+    context 'with show banned' do
+      it 'should assign users to all users' do
+        allow(User).to receive(:active)
```

Comment:
This is testing that `User` doesn't receive `active` -- the param essentially determines whether or not that scope is called